### PR TITLE
Default module

### PR DESCRIPTION
**Pull request by AlbyGiaco**

### CODE REVIEW

File: contracts/PositionManager.sol [hunk 1]
```diff
@@ -144,6 +151,23 @@ contract PositionManager is IPositionManager, ERC721Holder {
         return uniswapNFTsMemory;
     }
 
+    ///@notice set default data for every module
+    ///@param tokenId ID of the position
+    function setDefaultDataOfPosition(uint256 tokenId) internal {
```
Comment (RiccardoGalbusera) [inline contracts/PositionManager.sol:156]: _notation for internal functions?

**Review by RiccardoGalbusera — APPROVED** [review]
Also please delete newfixture.ts if it isn't needed

File: contracts/PositionManager.sol [hunk 2]
```diff
@@ -157,42 +181,32 @@ contract PositionManager is IPositionManager, ERC721Holder {
...
     ///@param moduleAddress address of the module
     ///@param data data for the module
     function setModuleData(
         uint256 tokenId,
         address moduleAddress,
-        bytes memory data
+        bytes32 data
     ) external override onlyOwner onlyOwnedPosition(tokenId) {
+        uint256 moduleData = uint256(data);
+        require(moduleData > 0, 'PositionManager::setModuleData: moduleData must be greater than 2%');
```
Comment (nicolas-racchi) [inline contracts/PositionManager.sol:194]: why greater than 2%?

File: contracts/PositionManager.sol [hunk 3]
```diff
@@ -157,42 +181,32 @@ contract PositionManager is IPositionManager, ERC721Holder {
...
-    ///@notice returns the data of a module strategy for tokenId position
-    ///@param tokenId ID of the position
-    ///@param moduleAddress address of the module
-    function getModuleData(uint256 tokenId, address moduleAddress)
+    ///@notice get info for a module strategy for tokenId position
+    ///@param _tokenId ID of the position
+    ///@param _moduleAddress address of the module
+    ///@return isActive is module activated
+    ///@return data of the module
+    function getModuleInfo(uint256 _tokenId, address _moduleAddress)
```
Comment (nicolas-racchi) [inline contracts/PositionManager.sol:203]: What is the advantage of calling getModuleInfo(id, address) vs activatedModules(id, address)?

File: contracts/Registry.sol [hunk 4]
```diff
@@ -84,18 +96,42 @@ contract Registry is IRegistry {
...
-        return modules[_id].contractAddress;
+    ///@param _defaultData default data for the module
+    function setDefaultValue(bytes32 _id, bytes32 _defaultData) external onlyGovernance {
+        require(modules[_id].contractAddress != address(0), 'Registry::setDefaultValue: Entry does not exist.');
+        modules[_id].defaultData = _defaultData;
     }
 
-    ///@notice Get the state of a module
+    function setDefaultActivation(bytes32 _id, bool _activatedByDefault) external onlyGovernance {
+        require(modules[_id].contractAddress != address(0), 'Registry::setDefaultValue: Entry does not exist.');
```
Comment (nicolas-racchi) [inline contracts/Registry.sol:108]: setDefaultActvation

File: contracts/utils/WithdrawRecipes.sol [hunk 5]
```diff
@@ -56,7 +56,15 @@ contract WithdrawRecipes {
     ///@notice remove a position from positionmanager zapping out
     ///@param tokenId ID of the token to withdraw
     ///@param tokenOut address of the token to withdraw
-    function zapOutUniNft(uint256 tokenId, address tokenOut) external {
+    function zapOutUniNft(uint256 tokenId, address tokenOut) external onlyOwner(tokenId) {
         IZapOut(positionManagerFactory.userToPositionManager(msg.sender)).zapOut(tokenId, tokenOut);
     }
+
+    modifier onlyOwner(uint256 tokenId) {
```
Comment (nicolas-racchi) [inline contracts/utils/WithdrawRecipes.sol:63]: Can we add a small message for this require?

File: contracts/Registry.sol (hunk 4 above)
Comment (nicolas-racchi) [reply contracts/Registry.sol:108]: + natspec

File: contracts/PositionManager.sol (hunk 3 above)
Comment (AlbyGiaco) [reply contracts/PositionManager.sol:203]: you cant call mapping externally using interface, if you want to call it you need to import directly the contract, waste of gas

**Pull request merged**